serialzy/serializers/base_model.py

```diff
@@ -69,13 +69,45 @@ def deserialize_from_dir(source: BinaryIO, read_from_dir) -> Any:
         return model
 
 
+def unpack_model_file(source: BinaryIO, destination: Path) -> None:
+    with destination.open("wb") as handle:
+        while True:
+            data = source.read(8096)
+            if not data:
+                break
+            handle.write(data)
+        handle.flush()
+        os.fsync(handle.fileno())
+
+
+def unpack_model_tar(source: BinaryIO, destination: Path) -> None:
+    with tempfile.NamedTemporaryFile(suffix=".tar") as handle:
+        while True:
+            data = source.read(8096)
+            if not data:
+                break
+            handle.write(data)
+        handle.flush()
+        os.fsync(handle.fileno())
+        shutil.unpack_archive(handle.name, destination, "tar")
```

Comment:
Actually, writing a file and than untaring it - is double disk job (because tar is already just sequence of files)
Maybe you should try to instantiate TarFile over source and unpack it instead of tmp file.

Reply:
это копипаста имеющегося кода